### ISSUE TITLE
Make path-style "as-needed" parameters not warn for strings with dots or brackets

### ISSUE THREAD

**Issue by itamarki** [wix/eslint-plugin-lodash#199]
Hi,

Thank you for this great plugin.

Configuration:
`"lodash/path-style": ["error", "as-needed"]`

consider the following code:
```javascript
const obj = {
    a: {
        "b.c": {
            d: 3
        }
    }
};

_.get(obj, ['a', 'b.c', 'd']); // -> 3
```

The rule will mark it as error and suggest to use a string for paths, but 
`_.get(obj, 'a.b.c.d');`
Is not equivalent code and will not work.

My suggestion is that "as-needed" configuration will not warn if the strings in the array contain dots or square brackets.

**Comment by itamarki** [comment]
After thinking it though, I found that this code can be written in string form:
`_.get(obj, 'a["b.c"].d');`
So the error message for `_.get(obj, ['a', 'b.c', 'd']);` is correct.

However I found another issue:
When using this rule with `--fix` on this code it will transform the code to:
`_.get(obj, 'a[b.c].d');` (note the missing quotes around `b.c`).

**Comment by snoozbuster** [comment]
Related bug with the `--fix` option:
 
With options `[2, "as-needed"]`, this rule suggests the following diff:

```
- lodashGet(obj, ['E', `foo_${a}_${b}`])
+lodashGet(obj, `E${`foo_${a}_${b}`}`)
```

This is, of course, incorrect (there is no `.` between E and the first template string) - but furthermore, imo template strings should really be considered as variables and this suggestion shouldn't be made in the first place.
